src/dependency_injector/_cwiring.pyx

```diff
@@ -1,83 +1,109 @@
 """Wiring optimizations module."""
 
-import asyncio
-import collections.abc
-import inspect
-import types
+from asyncio import gather
+from collections.abc import Awaitable
+from inspect import CO_ITERABLE_COROUTINE
+from types import CoroutineType, GeneratorType
 
+from .providers cimport Provider, Resource, NULL_AWAITABLE
 from .wiring import _Marker
 
-from .providers cimport Provider, Resource
+cimport cython
 
 
-def _sync_inject(object fn, tuple args, dict kwargs, dict injections, dict closings, /):
-    cdef object result
+@cython.no_gc
```

Comment:
@ZipFile does it only disable the check of circular dependencies in itself or do we need to clean up these objects on our own now?

Reply:
Yes, It [disables circular checks](https://cython.readthedocs.io/en/latest/src/userguide/extension_types.html#disabling-cyclic-garbage-collection) only (skips `tp_traverse` and `tp_clear` slots). Since this is an internal class, we're confident enough that it'll never reference itself, even indirectly.

Here's full diff with and without `@cython.no_gc` deco: [no_gc.patch](https://github.com/user-attachments/files/20577371/no_gc.patch)